### PR TITLE
rayjob controller: refactor environment variable check in unit tests

### PR DESCRIPTION
**Pull request by andrewsykim**
## Why are these changes needed?

Very minor refactor based on https://github.com/ray-project/kuberay/pull/1852#discussion_r1462194366. 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [X] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/controllers/ray/utils/util.go [hunk 1]
```diff
@@ -512,3 +512,14 @@ func EnvVarExists(envName string, envVars []corev1.EnvVar) bool {
 	}
 	return false
 }
+
+// EnvVarByName returns an entry in []corev1.EnvVar that matches a name.
```
[the remark this answers is not in the document]
Comment (andrewsykim) [reply ray-operator/controllers/ray/utils/util.go:516]: I don't think it hurts to have both. EnvVarExists returning only a bool is a bit cleaner in some cases:

For example:
```
if !utils.EnvVarExists(utils.RAY_PORT, container.Env) {
    portEnv := corev1.EnvVar{Name: utils.RAY_PORT, Value: headPort}
    container.Env = append(container.Env, portEnv)
}
``` 

vs

```
if _, exists := !utils.EnvVarExists(utils.RAY_PORT, container.Env); !exists {
    portEnv := corev1.EnvVar{Name: utils.RAY_PORT, Value: headPort}
    container.Env = append(container.Env, portEnv)
}
```

[the remark this answers is not in the document]
Comment (andrewsykim) [reply ray-operator/controllers/ray/utils/util.go:516]: I don't feel strongly about it though, I can replace all calls to EnvVarExists with EnvVarByName if you think it's worthwhile

**Comment by kevin85421** [comment]
I will review this PR after #1868 is merged.

**Comment by andrewsykim** [comment]
Rebased after https://github.com/ray-project/kuberay/pull/1868 merged

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**